#!/usr/bin/env python
PACKAGE = "bumblebob_lateral_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("lateral_gain", double_t, 0, "The gain parameter for the lateral value",  1.0, 0, 10)
gen.add("heading_gain", double_t, 0, "The gain parameter for the heading value",  1.2, 0, 10)
gen.add("delta_dot_gain", double_t, 0, "The gain parameter for the change of steering value",  0.3, 0, 10)

gen.add("look_front", int_t, 0, "The estimation for the area in front of the car ",  10, 0, 20)
gen.add("look_back", int_t, 0, "The estimation for the area behind the car ",  10, 0, 20)


exit(gen.generate(PACKAGE, "bumblebob_lateral_controller", "Dynamic"))
